laurat007/assignment3/Trie.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+
+/**
+ *
+ * @author Laura
+ *  Implemented a Trie structure to build a dictionary.
+ *  The Trie works like this: when a word is added, the trie instantiates 
+ * its nth child (n = first letter of the word - 'A' in ASCII codes), so it 
+ * goes increasing the height of the trie, simultaneously with the length 
+ * of the word. If the nth child is already instantiated, the algorithm goes
+ * to the next step.
+ *  
+ */
+ 
+public class Trie {
+     
+    static final int MAX_LETTERS = 26;
+    
+    class TrieNode
+    {
+        TrieNode[] children = new TrieNode[MAX_LETTERS];
```

Comment:
I think it would be better to use map here: you don't have to allocate space for 26 children for all the nodes that are created, this wastes a lot of memory for nothing.

Reply:
done. 
It was the first thing that came to my mind to build it like that and I didn't consider changing it after finishing.